packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/route.spec.js

```diff
@@ -96,15 +126,189 @@ describe('licence-summary > route', () => {
 
     it('should return a redirect error if lastName is not included on the licensee', async () => {
       mockTransactionCacheGet.mockImplementationOnce(() => ({ licensee: { firstName: 'John' } }))
+      const transaction = {
+        permissions: {
+          length: 0,
+          isLicenceForYou: false
+        }
+      }
       let error = false
       try {
-        await getData(mockRequest)
+        await getData(mockRequest(transaction))
       } catch (e) {
         error = e
       }
       expect(error).not.toBeFalsy()
       expect(error).toBeInstanceOf(GetDataRedirect)
       expect(error.redirectUrl).toBe(NAME.uri)
     })
+
+    it('should return a redirect error if date of birth is not included on the licensee', async () => {
+      mockTransactionCacheGet.mockImplementationOnce(() => ({ licensee: { firstName: 'John', lastName: 'Smith' } }))
+      const transaction = {
+        permissions: {
+          length: 0,
+          isLicenceForYou: false
+        }
+      }
+      let error = false
+      try {
+        await getData(mockRequest(transaction))
+      } catch (e) {
+        error = e
+      }
+      expect(error).not.toBeFalsy()
+      expect(error).toBeInstanceOf(GetDataRedirect)
+      expect(error.redirectUrl).toBe(DATE_OF_BIRTH.uri)
+    })
+
+    it('should return a redirect error if start date is not included on the licensee', async () => {
+      mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' }
+      }))
+      const transaction = {
+        permissions: {
+          length: 0,
+          isLicenceForYou: false
+        }
+      }
+      let error = false
+      try {
+        await getData(mockRequest(transaction))
+      } catch (e) {
+        error = e
+      }
+      expect(error).not.toBeFalsy()
+      expect(error).toBeInstanceOf(GetDataRedirect)
+      expect(error.redirectUrl).toBe(LICENCE_TO_START.uri)
+    })
+
+    it('should return a redirect error if number of rods is not included on the licensee', async () => {
+      mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
+        licenceStartDate: '2025-01-01',
+        licenceType: 'Salmon and sea trout'
+      }))
+      const transaction = {
+        permissions: {
+          isLicenceForYou: false
+        }
+      }
+      let error = false
+      try {
+        await getData(mockRequest(transaction))
```

Comment:
you're mocking the transaction get as well as passing in a mocked transaction. I think your best bet is to pass in a mocked transaction, but do only that rather than also using mockTransactionCacheGet in all the tests.